sentry-android-core/src/main/java/io/sentry/android/core/TempSensorBreadcrumbsIntegration.java

```diff
@@ -0,0 +1,98 @@
+package io.sentry.android.core;
+
+import static android.content.Context.SENSOR_SERVICE;
+
+import android.content.Context;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import androidx.annotation.Nullable;
+import io.sentry.core.Breadcrumb;
+import io.sentry.core.IHub;
+import io.sentry.core.Integration;
+import io.sentry.core.SentryLevel;
+import io.sentry.core.SentryOptions;
+import io.sentry.core.util.Objects;
+import java.io.Closeable;
+import java.io.IOException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.TestOnly;
+
+public final class TempSensorBreadcrumbsIntegration
+    implements Integration, Closeable, SensorEventListener {
+
+  private final @NotNull Context context;
+  private @Nullable IHub hub;
+  private @Nullable SentryAndroidOptions options;
+
+  @TestOnly @Nullable SensorManager sensorManager;
+
+  public TempSensorBreadcrumbsIntegration(final @NotNull Context context) {
+    this.context = Objects.requireNonNull(context, "Context is required");
+  }
+
+  @SuppressWarnings("deprecation")
+  @Override
+  public void register(final @NotNull IHub hub, final @NotNull SentryOptions options) {
+    this.hub = Objects.requireNonNull(hub, "Hub is required");
+    this.options =
+        Objects.requireNonNull(
+            (options instanceof SentryAndroidOptions) ? (SentryAndroidOptions) options : null,
+            "SentryAndroidOptions is required");
+
+    this.options
+        .getLogger()
+        .log(
+            SentryLevel.DEBUG,
+            "enableSystemEventsBreadcrumbs enabled: %s",
+            this.options.isEnableSystemEventBreadcrumbs());
+
+    if (this.options.isEnableSystemEventBreadcrumbs())
+      sensorManager = (SensorManager) context.getSystemService(SENSOR_SERVICE);
+    if (sensorManager == null) {
+      this.options.getLogger().log(SentryLevel.INFO, "SENSOR_SERVICE is not available.");
```

Comment:
I believe this was a typo or IDE change not sure, ideally the 1st `if` would take the whole block or would not make sense.
it's there since my 1st commit, damn, and tests passed haha